crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,256 @@
+};
 
-pub(crate) fn find_attr(attrs: &[Attribute], name: &str) -> Option<Meta> {
-    attrs.iter().find_map(|a| {
-        if let Ok(meta) = a.parse_meta() {
-            if meta.path().is_ident(name) {
-                Some(meta)
-            } else {
-                //println!("attr name: {}",meta.name());
-                None
+use crate::ast::prop::PropAttrsType;
+
+/// Parses the specified attributes from a `syn::Attribute` iterator.
+/// # Arguments
+/// * `attrs` - `&[Attribute]` iterator
+/// * `attr_ident` - The path ident to search for
+/// * `meta` - a return variable that will give us the parse_nested_meta of each element
+/// * `field` - nested path to search for
+/// * `opt` - mutable variable to save data into
+/// * `block` - block of code to execute
+macro_rules! parse_attributes {
+    ($attrs:expr, $attr_ident:expr, $meta: ident, $($field:pat, $opt:expr => $block:expr)*) => {
+        for attr in $attrs {
+            if !attr.path().is_ident($attr_ident) {
+                continue;
             }
-        } else {
-            //println!("unrecog attribute");
-            None
+
+            attr.parse_nested_meta(|$meta| {
+
+                let ident = $meta.path.get_ident().ok_or_else(|| $meta.error(concat!("unrecognized ", $attr_ident, " attribute")))?;
+                let attr_name = &ident.to_string();
+
+                match attr_name.as_str() {
+                    $(
+                        $field if $opt.is_none() => {
+                            return Ok($block)
+                        },
+                        $field => {return Err($meta.error(concat!($attr_ident, " attribute already specified")))}
+                    )*
+
+                    _ => return Err($meta.error(concat!("unrecognized ", $attr_ident, " attribute"))),
+                }
+            })?;
         }
-    })
+    };
 }
 
-pub(crate) fn find_name_attribute<'a>(meta: &'a Meta, name: &str) -> Option<&'a MetaNameValue> {
-    find_meta(meta, name).map(|meta| match meta {
-        Meta::NameValue(name_value) => name_value,
-        _ => panic!("should not happen"),
-    })
+// This piece of code can be included in the macro but there are some issues
+// For example if we return this from the parse_attributes! macro with closure function
+// It works but there is an issue with clippy giving a "warning: try not to call a closure in the expression where it is declared"
+// Here is how this can be fixed if used in macro: https://stackoverflow.com/questions/76989679/how-to-call-a-closure-from-a-macro
+// So to do this in a less work around way we use a separate function like so:
+// We return $meta variable like shown here: https://github.com/rust-lang/rust-clippy/issues/1553
+pub fn parse_attributes_data(meta: &ParseNestedMeta) -> (Option<syn::Expr>, Span, String) {
+    // we can safely unwarp as this is already checked from the parse_attributes macro
+    let ident = meta.path.get_ident().unwrap();
```

Comment:
Even if it's true for now, it might change later. This is a public function in `utils` crate, so `unwrap` and ignoring `Err` results for functions should not be introduced.

Reply:
The comment is still about the old version.

And, the result of `value.parse()` below is ignored.